### PR TITLE
Able to view sheep and shepherd attendees at each event

### PR DESCRIPTION
**Pull request by stevenshuhyow**

### CODE REVIEW

File: server/routes.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ module.exports = function(app, express, passport) {
 
   app.use('/api/event', require('./api/event'));
   app.use('/api/observation', require('./api/observation'));
+  app.use('/api/attendees', require('./api/attendee'));
```
Comment (kevhuang) [inline server/routes.js]: Any objections to using /api/attendee (singular) for the base path just to be consistent with the other paths?

File: client/public/js/actions/ObservationActions.js [hunk 2]
```diff
@@ -14,7 +14,9 @@ var ObservationActions = {
    * @param {number} eventId The primary key of the event in the database
    */
   getAllByEvent: function(eventId) {
+    // console.log(eventId, 'eventID inside of observationActions');
```
Comment (kevhuang) [inline client/public/js/actions/ObservationActions.js:17]: Remove comments if they're not being used

**Comment by kevhuang** [comment]
Let's refactor to have the list of sheep and shepherds rendered from within the event detail component instead of the observation list component.

File: client/public/js/actions/ObservationActions.js [hunk 3]
```diff
@@ -14,7 +14,9 @@ var ObservationActions = {
...
     ObservationAPI.getAllObservationsByEvent(eventId).then(function(observations) {
+      // console.log(observations, "observations inside of observationActions");
```
Comment (ekong2) [inline client/public/js/actions/ObservationActions.js:19]: ^

File: client/public/js/components/AttendeesList.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  getInitialState: function() {
+    return {
+      // Initialize empty set of attendees on initial render
+      sheeps: [],
+      shepherds: []
+    };
+  },
+
+  componentDidMount: function() {
+    AttendeeStore.addEventListener('aquiredShepherds', this._acquiredShepherds);
```
Comment (ekong2) [inline client/public/js/components/AttendeesList.js]: typo 'aquired'

File: client/public/js/components/AttendeesList.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    // Fetching via AJAX needs to happen after mounting due to async
+    if (this.isMounted()) {
+      AttendeeActions.getAllSheepsByEvent(this.props.eventId);
+      AttendeeActions.getAllShepherdsByEvent(this.props.eventId);
+    }
+  },
+
+  componentWillUnmount: function() {
+    // Remove event listeners when the DOM element is removed
+    AttendeeStore.removeEventListener('aquiredShepherds', this._acquiredShepherds);
```
Comment (ekong2) [inline client/public/js/components/AttendeesList.js]: 'aquired'

File: client/public/js/components/AttendeesList.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    AttendeeStore.removeEventListener('acquiredSheep', this._acquiredSheeps);
+
+  },
+
+  render: function() {
+    //referenced the observations retrieved from EventDetailStore using getAllObservations()
+    //sends each observation to AttendeesListItem so that it will be properly
+    //displayed
+
+    var shepherds = this.state.shepherds.map(function(shepherd, i) {
```
Comment (ekong2) [inline client/public/js/components/AttendeesList.js:52]: no need for 'i'

File: client/public/js/components/AttendeesList.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      // console.log(shepherd, 'inside of render AttendeesList')
+      return <ShepherdListItem key={shepherd.id} shepherd={shepherd}/>
+    });
+
+    var sheeps = this.state.sheeps.map(function(sheep, i) {
```
Comment (ekong2) [inline client/public/js/components/AttendeesList.js:57]: no need for i

File: client/public/js/stores/AppStore.js [hunk 8]
```diff
@@ -54,12 +54,12 @@ AppDispatcher.register(function(payload) {
     case AppConstants.SET_CURRENT_USER:
       _currentUser.id = payload.user.id;
       _currentUser.username = payload.user.username;
-      console.log(payload, 'payloaddd');
+      // console.log(payload, 'payloaddd');
```
Comment (ekong2) [inline client/public/js/stores/AppStore.js:57]: get rid of comments

File: client/public/js/stores/AppStore.js [hunk 9]
```diff
@@ -54,12 +54,12 @@ AppDispatcher.register(function(payload) {
...
       AppStore.emitEvent('loggedIn');
       break;
 
     case AppConstants.REMOVE_CURRENT_USER:
-      console.log('removing user inside of appstore.js');
+      // console.log('removing user inside of appstore.js');
```
Comment (ekong2) [inline client/public/js/stores/AppStore.js:62]: get rid of comments? up to you if you still need them

File: client/public/js/stores/AttendeeStore.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+// Register callback to handle all updates
+AppDispatcher.register(function(payload) {
+  switch(payload.actionType) {
+    // When sheep are fetched
+    case AppConstants.SHEPHERDS_GET:
+      // Set current sheep collection to the fetched results
+      _shepherds = payload.shepherds;
+      // console.log(_shepherds, 'inside of attendee store!!')
+      AttendeeStore.emitEvent('aquiredShepherds');
```
Comment (ekong2) [inline client/public/js/stores/AttendeeStore.js]: typo aquired

File: client/public/js/stores/AttendeeStore.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+      break;
+
+    case AppConstants.SHEEPS_GET:
+      // Set current sheep collection to the fetched results
+      _sheeps = payload.sheeps;
+      // console.log(_sheeps, 'inside of attendee store!!')
+      AttendeeStore.emitEvent('aquiredSheep');
```
Comment (ekong2) [inline client/public/js/stores/AttendeeStore.js]: typo aquired

File: client/public/js/stores/ObservationStore.js [hunk 12]
```diff
@@ -44,6 +44,7 @@ AppDispatcher.register(function(payload) {
     case AppConstants.OBSERVATION_GET:
       // Set current observations collection to the fetched results
       _observations = payload.observations;
+      // console.log(_observations, "observations inside of ObservationStore");
```
Comment (ekong2) [inline client/public/js/stores/ObservationStore.js:47]: get rid of comments

File: client/public/js/utils/AttendeeAPI.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+ * Getter and setter functions for interacting with the server event API endpoints
+ */
+
+var AttendeesAPI = {
+  /**
+   * Get all attendees for an event
+   * @param {number} eventId The primary key of the event in the database
+   */
+  getAllShepherdsByEvent: function(eventId) {
+    // console.log(eventId, 'inside of getAllAttendeesByEvent');
```
Comment (ekong2) [inline client/public/js/utils/AttendeeAPI.js:11]: ^

File: client/public/js/utils/AttendeeAPI.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    // Return a promise so the requester can chain the response
+    return $.ajax({
+      url: '/api/attendees/shepherd/' + eventId,
+      method: 'GET',
+      dataType: 'json'
+    }).then(function(data) {
+      // console.log('ShepherdAPI', data);
```
Comment (ekong2) [inline client/public/js/utils/AttendeeAPI.js:18]: ^

File: client/public/js/utils/AttendeeAPI.js [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+   getAllSheepsByEvent: function(eventId) {
+    // console.log(eventId, 'inside of getAllAttendeesByEvent');
+    // Return a promise so the requester can chain the response
+    return $.ajax({
+      url: '/api/attendees/sheep/' + eventId,
+      method: 'GET',
+      dataType: 'json'
+    }).then(function(data) {
+      // console.log('SheepAPI', data);
```
Comment (ekong2) [inline client/public/js/utils/AttendeeAPI.js:35]: ^

File: client/public/js/utils/AttendeeAPI.js [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+      // Return attendees on success
+      return data;
+    }, function() {
+      // Return empty array on error
+      return [];
+    });
+  },
+
+   getAllSheepsByEvent: function(eventId) {
+    // console.log(eventId, 'inside of getAllAttendeesByEvent');
```
Comment (ekong2) [inline client/public/js/utils/AttendeeAPI.js:28]: ^

File: server/routes.js (hunk 1 above)
Comment (ekong2) [inline server/routes.js]: agree, no objections

**Comment by kevhuang** [comment]
Good

**Comment by ekong2** [comment]
:+1: nice job

**Pull request merged**